examples/stable/mysql-csi-snapshot/README.md

```diff
@@ -0,0 +1,269 @@
+```
+
+## Disaster strikes!
+
+Let's say someone accidentally deleted the test database.
+
+```bash
+# Connect to MySQL by running a shell inside MySQL's pod
+$ kubectl exec -it $(kubectl get pods -n mysql --selector=app.kubernetes.io/instance=mysql-release -o=jsonpath='{.items[0].metadata.name}') -n mysql -- bash
+
+$ mysql --user=root --password=asd#45@mysqlEXAMPLE
+
+mysql> SHOW DATABASES;
++--------------------+
+| Database           |
++--------------------+
+| information_schema |
+| my_database        |
+| mysql              |
+| performance_schema |
+| sys                |
+| test               |
++--------------------+
+6 rows in set (0.00 sec)
+
+mysql> DROP DATABASE test;
+Query OK, 1 row affected (0.12 sec)
+
+mysql> SHOW DATABASES;
++--------------------+
+| Database           |
++--------------------+
+| information_schema |
+| my_database        |
+| mysql              |
+| performance_schema |
+| sys                |
++--------------------+
+5 rows in set (0.01 sec)
+```
+
+## Restore MySQL DB
+
+To restore the missing data, you should use the backup that you created before. We use `kanctl` for creating this restore action. `kanctl` helps create ActionSets that depend on other ActionSets.
+
+```bash
+# Make sure to use correct backup actionset name here
+$ kanctl --namespace kanister create actionset --action restore --from backup-mlvcv
+actionset restore-backup-mlvcv-6z9xn created
+
+# Check the PVCs in mysql namespace. You should see a new PVC in the format '<PVC-name>-restored'
+$ kubectl -n mysql  get pvc
+NAME                            STATUS   VOLUME                                     CAPACITY   ACCESS MODES   STORAGECLASS       AGE
+data-mysql-release-0            Bound    pvc-b966c5ef-3d20-4ec2-9ab3-1a868737ea0a   8Gi        RWO            do-block-storage   4m14s
+data-mysql-release-0-restored   Bound    pvc-2eb6fb89-78bc-4c2e-8c1e-c270d664f311   8Gi        RWO            do-block-storage   18s
+
+# Use following command to debug if a new PVC in the format '<PVC-name>-restored' is not created
+$ kubectl --namespace kanister describe actionset restore-backup-mlvcv-6z9xn
+```
+
+Next, we need to add the newly restored PVC `data-mysql-release-0-restored` in the `mysql-release` StatefulSet spec template. One of the easiest way you can do that is by using `kubectl patch` command.
+
+```bash
+$ kubectl -n mysql patch statefulset mysql-release --type=json -p='[{"op": "add", "path": "/spec/template/spec/volumes/-", "value": {"name": "restored-data", "persistentVolumeClaim": {"claimName": "data-mysql-release-0-restored"}}}]'
+
+$ kubectl -n mysql patch statefulset mysql-release --type=json -p='[{"op": "add", "path": "/spec/template/spec/containers/0/volumeMounts", "value": [{"mountPath": "/bitnami/mysql", "name": "restored-data"}, {"mountPath": "/opt/bitnami/mysql/conf/my.cnf", "name": "config", "subPath": "my.cnf"}]}]'
```

Comment:
should these be statefulsets or deployments.

Reply:
These should be statefulsets